resources/inspec.rb

```diff
@@ -22,6 +29,49 @@
   end
 end
 
+action :upload do
+  converge_by 'run profile validation checks' do
+    raise 'Path to profile archive not specified' if path.nil?
+    raise "Profile archive file #{path} does not exist." unless ::File.exist?(path)
+    profile = Inspec::Profile.for_target(path, {})
+    error_count = 0
+    lambda { |msg|
+      error_count += 1
+      Chef::Log.error msg
+    }
+    result = profile.check
+    Chef::Log.info result[:summary].inspect
+    raise 'Profile check failed' unless result[:summary][:valid]
+    Chef::Log.info 'Profile is valid'
+  end
+
+  converge_by 'upload compliance profile' do
+    require 'bundles/inspec-compliance/configuration'
+    require 'bundles/inspec-compliance/support'
+    require 'bundles/inspec-compliance/http'
+    require 'bundles/inspec-compliance/api'
+    require 'bundles/inspec-compliance/target'
+    _success, msg, access_token = Compliance::API.get_token_via_refresh_token(server, refresh_token, true)
+
```

Comment:
This was solved in the past since there was a compliance library.  All libraries get compiled and automatically included in the cookbook resources.  That would perhaps be a decent place for all the requires and the compliance API methods.